markdown/ch3.markdown

```diff
@@ -724,38 +800,43 @@ survival = 1 - cumfreq
 Then, plot using Matplotlib:
 
 ```python
-plt.loglog(np.arange(1, len(survival) + 1), survival)
-plt.xlabel('in-degree distribution')
-plt.ylabel('fraction of neurons with higher in-degree distribution')
-plt.scatter(avg_in_degree, 0.0022, marker='v')
-plt.text(avg_in_degree - 0.5, 0.003, 'mean=%.2f' % avg_in_degree)
-plt.ylim(0.002, 1.0)
-plt.show()
+fig, ax = plt.subplots()
+ax.loglog(np.arange(1, len(survival) + 1), survival)
+ax.set_xlabel('in-degree distribution')
+ax.set_ylabel('fraction of neurons with higher in-degree distribution')
+ax.scatter(avg_in_degree, 0.0022, marker='v')
+ax.text(avg_in_degree - 0.5, 0.003, 'mean=%.2f' % avg_in_degree)
+ax.set_ylim(0.002, 1.0)
 ```
 
 <!-- exercise begin -->
 
-**Exercise:** Use `scipy.optimize.curve_fit` to fit the tail of the
+### Exercise: Complete Figure 6B from the paper
+
+This exercise is a bit of a preview for chapter 7 (optimization):
+use `scipy.optimize.curve_fit` to fit the tail of the
 in-degree survival function to a power-law,
 $f(d) \sim d^{-\gamma}, d > d_0$,
 for $d_0 = 10$ (the red line in Figure 6B of the paper), and modify the plot
 to include that line.
 
 <!-- exercise end -->
 
+You now should have a fundamental understanding of graphs as a scientific
+abstraction, and how to easily manipulate and analyse them using Python and
+NetworkX. Now, we move on to a particular kind of graph used in image
+processing and computer vision.
+
 ## Region adjacency graphs
```

Comment:
If this heading were something like "Region adjacency graphs for image segmentation" that might address Nan's comment about a segmentation subheading just below.